lib/cluster/subscriber.js

```diff
@@ -680,7 +680,15 @@ class ClusterSubscriber {
    * to recover, otherwise we evict it from the cluster.
    */
   async checkHeartbeat() {
-    if (this.state === stateEnum.EVICTED || this.remoteNodeEvictionPrevented) {
+    if (this.remoteNodeEvictionPrevented) {
+      // Fake the heartbeat while the node eviction prevention is enabled
+      // otherwise when the node eviction prevention is disabled
+      // the node will be evicted if it did not send a heartbeat before disabling the protection.
+      this.lastHeartbeat = Date.now();
+      return;
+    }
+
+    if (this.state === stateEnum.EVICTED) {
```

Comment:
Instead of faking the heartbeat in order to prevent the node to kill itself, could we add a new special state in stateEnum like `DEBUGGING` to ignore a missing heartbeat in the condition branches below?

Reply:
It's already the case, but when exiting this special state all other nodes that had not received the heartbeat round from the node that was debugging will evict it if the node that was previously used to debug did not have the opportunity to send a heartbeat before their check, that's why we are faking it meanwhile, this way the node will have the time to send a proper heartbeat when exiting his state.